kore/src/Kore/JsonRpc.hs

```diff
@@ -123,13 +127,34 @@ respond serverState moduleName runSMT =
             toStopLabels cpRs tRs =
                 Exec.StopLabels (fromMaybe [] cpRs) (fromMaybe [] tRs)
 
+            containsLabelOrRuleId rules = \case
+                Nothing -> Nothing
+                Just lblsOrRuleIds ->
+                    let requestSet =
+                            Set.fromList $ concat [[Left $ Label $ Just lblOrRid, Right $ UniqueId $ Just lblOrRid] | lblOrRid <- lblsOrRuleIds]
+                        ruleSet =
+                            Set.fromList $ concat [[Left lbl, Right ruleId] | ((ruleId, lbl), _) <- toList rules]
+                     in either unLabel getUniqueId <$> Set.lookupMin (requestSet `Set.intersection` ruleSet)
+            mkLogs rules =
+                concat
+                    [ [ Simplification
+                        Nothing
+                        (Success Nothing (fromMaybe "UNKNOWN" $ getUniqueId s))
+                        KoreRpc
+                      | fromMaybe False logSuccessfulSimplifications
+                      , s <- toList simps
+                      ]
+                        ++ [Rewrite (Success Nothing (fromMaybe "UNKNOWN" $ getUniqueId r)) KoreRpc | fromMaybe False logSuccessfulRewrites]
+                    | ((r, _), simps) <- toList rules
+                    ]
```

Comment:
IIUC this will always traverse the list of rules from the result, but omits the 
Maybe better to guard with `logSuccessfulSimplifications, logSuccessFulRewrites` on the outside so we avoid the overhead when no logs at all were requested. Something like
```suggestion
            mkLogs rules
              | logSuccessfulSimplifications || logSuccessfulRewrites =
                Just . concat $
                    [ [ Simplification
                        Nothing
                        (Success Nothing (fromMaybe "UNKNOWN" $ getUniqueId s))
                        KoreRpc
                      | fromMaybe False logSuccessfulSimplifications
                      , s <- toList simps
                      ]
                        ++ [Rewrite (Success Nothing (fromMaybe "UNKNOWN" $ getUniqueId r)) KoreRpc | fromMaybe False logSuccessfulRewrites]
                    | ((r, _), simps) <- toList rules
                    ]
              | otherwise = Nothing
```
(This also changes the type to a `Maybe [LogEntry]` as suggested above)